pkg/cmd/roachtest/test_runner.go

```diff
@@ -1563,6 +1585,10 @@ func (r *testRunner) teardownTest(
 		getGoCoverArtifacts(ctx, c, t)
 	}
 
+	// Terminate tasks to ensure that any stray tasks are cleaned up.
+	t.L().Printf("terminating tasks")
+	t.taskManager.Terminate(t.L())
```

Comment:
Is there a reason we only want to terminate task on successful tests? If not, should we move this to a defer at the top? Otherwise it's possible we return before we reach here:

https://github.com/cockroachdb/cockroach/blob/9ec5ed022888460e8e808c18a7a96ad742882512/pkg/cmd/roachtest/test_runner.go#L1574

Reply:
Nice catch, didn't notice the short-circuit above! I'll amend.